crypto/curve25519/curve25519.c

```diff
@@ -1994,7 +1995,7 @@ void ED25519_keypair_from_seed(uint8_t out_public_key[32],
   x25519_ge_scalarmult_base(&A, az);
   ge_p3_tobytes(out_public_key, &A);
 
-  OPENSSL_memcpy(out_private_key, seed, 32);
+  OPENSSL_memcpy(out_private_key, seed, ED25519_SEED_LEN);
   OPENSSL_memcpy(out_private_key + 32, out_public_key, 32);
```

Comment:
There's a "slippery slope" concern here in that it's not clear which/how-many of the *magic numbers* (32 and 64) in the logic here should be replaced by their corresponding constants (ED25519_SEED_LEN, ED25519_PUBLIC_KEY_LEN, ED25519_PRIVATE_KEY_LEN and ED25519_SIGNATURE_LEN) for the current PR.

I think a reasonable line to draw is to replace "32" with "ED25519_SEED_LEN" wherever appropriate, but otherwise leave the other magic numbers alone for now.
```
  OPENSSL_memcpy(out_private_key, seed, ED25519_SEED_LEN);
  OPENSSL_memcpy(out_private_key + ED25519_SEED_LEN, out_public_key, 32);
```